### PR TITLE
chore: Add ability to use gitlab for building.

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

On Android, this is split between a build on a Linux machine (using a docker file) and a macOS machine for integration testing.  This also adds scripts to create and launch Android emulators.

On iOS, we use the macOS runners for building and integration testing.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+include:
+  - 'https://gitlab-templates.ddbuild.io/slack-notifier/v1/template.yml'
+
+variables:
+  CURRENT_CI_IMAGE: "5"
+  BUILD_STABLE_REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
```
Comment (0xnm) [inline .gitlab-ci.yml:6]: I think it can be `registry.ddbuild.io` (but not sure if it makes a lot of difference)

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    - dart pub global activate junitreport
+    - melos bootstrap
+    - mkdir -p $CI_PROJECT_DIR/.build/test-results/
+    - melos prepare
+
+.pre-ios:
+  script:
+    - xcode-select -p
+    - xcodebuild -runFirstLaunch
+    - xcodebuild -downloadPlatform iOS
```
Comment (0xnm) [inline .gitlab-ci.yml:33]: why this is needed? it should already come with the runner, no?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  script:
+    - docker buildx build --tag registry.ddbuild.io/${CI_IMAGE_REPO}:${CURRENT_CI_IMAGE} --label target=build -f Dockerfile.gitlab --push .
+
+# Build (And Analyze) Stage
+
+build-flutter:
+  stage: build
+  tags:
+    - macos:sonoma
+    - specific:true
```
Comment (0xnm) [inline .gitlab-ci.yml:55]: is this one needed?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  tags: 
+    - macos:sonoma
+    - specific:true
+  script:
+    - !reference [.pre, script]
+    - melos pub:get
+    - yes | $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager "build-tools;33.0.3" "platform-tools" "platforms;android-33" || true
+    - yes | flutter doctor --android-licenses || true
+    - flutter doctor
+    - cd tools/ci && dart pub get && dart run ci_helpers start_sim --platform android --sdk "33"
```
Comment (0xnm) [inline .gitlab-ci.yml:105]: side note: we should make sure we target API 34 in tests.  Moreover, API 35 will be released soon.

Update: there is also an inconsistency with `Dockerfile`, which has things set up for API 34.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    - melos run analyze:ios
+    - melos run build:ios
+    - melos run unit_test:ios    
+
+# Integration Tests
+
+android-integration-test:
+  stage: integration-test  
+  tags: 
+    - macos:sonoma
```
Comment (0xnm) [inline .gitlab-ci.yml:97]: shouldn't it be just a linux box for Android? 🤔  Or is it related to the virtualization?

**Review by 0xnm — COMMENTED** [review]
nice to see that move! I left few questions.

File: .gitlab-ci.yml (hunk 1 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:6]: I'll double check. I seem to remember `registry` didn't work but I forget why.

File: .gitlab-ci.yml (hunk 2 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:33]: I think it's supposed to, but the Flutter image doesn't have it setup sometimes? It's more a safety thing, and it fast exits if the platform is already downloaded.

File: .gitlab-ci.yml (hunk 3 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:55]: Yes. The macOS runners are set up differently for different repos, and this tells gitlab to use the one specific to this repo.

File: .gitlab-ci.yml (hunk 4 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:105]: 👍 I'll update the integration tests to 34.

File: .gitlab-ci.yml (hunk 5 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:97]: Integration tests require an emulator, which we can't figure out how to run virtualized, hence why this is running on the macOS runner.

**Review by 0xnm — DISMISSED** [review]

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  artifacts:
+    when: always
+    expire_in: "30 days"
+    reports:
+      junit: $CI_PROJECT_DIR/.build/test-results/*.xml
+
+# web-integration-test:
+#   stage: integration-test
+#   script:
+#     - melos run integration_test:web
```
Comment (0xnm) [inline .gitlab-ci.yml:137]: should it be kept commented?

**Review by 0xnm — APPROVED** [review]

File: .gitlab-ci.yml (hunk 6 above)
Comment (fuzzybinary) [reply .gitlab-ci.yml:137]: Yes for now. It's my next task.

**Pull request merged**